admin/downloads_manager.php

```diff
@@ -179,10 +187,22 @@
                   <td><?php echo zen_values_name($products_downloads['options_values_id']); ?></td>
                   <td>
                     <div class="fa-stack fa-fw">
-                      <i class="fa fa-circle fa-stack-1x<?php echo (!zen_orders_products_downloads($products_downloads['products_attributes_filename']) ? ' txt-red' : ' txt-lime'); ?>"></i>
+                      <i class="fa fa-circle fa-stack-1x<?php echo (!$file_check ? ' txt-red' : ' txt-lime'); ?>"></i>
                       <i class="fa fa-circle-o fa-stack-1x txt-black"></i>
                     </div>
                     <?php echo $products_downloads['products_attributes_filename']; ?></td>
+<?php if ($show_download_date) { 
+         $file_date = ''; 
+         if ($file_check) { 
+            $handler = zen_get_download_handler($products_downloads['products_attributes_filename']); 
+        
+            if ($handler == 'local') {
+              $file_date = $zcDate->output(DATE_TIME_FORMAT, filemtime(DIR_FS_DOWNLOAD . $products_downloads['products_attributes_filename']));
```

Comment:
Do we "know" that the file exists at this point?  If not, that's going to be a PHP warning (at best).

Reply:
Yes because of `$file_check`.  For local handlers, returns `file_exists`.